android/src/main/java/com/amplitude/android/DefaultTrackingOptions.kt

```diff
@@ -1,27 +1,34 @@
 package com.amplitude.android
 
-open class DefaultTrackingOptions @JvmOverloads constructor(
-    var sessions: Boolean = true,
-    var appLifecycles: Boolean = false,
-    var deepLinks: Boolean = false,
-    var screenViews: Boolean = false
-) {
-    // Prebuilt options for easier usage
-    companion object {
-        @JvmField
-        val ALL = DefaultTrackingOptions(
-            sessions = true,
-            appLifecycles = true,
-            deepLinks = true,
-            screenViews = true
-        )
+open class DefaultTrackingOptions
+    @JvmOverloads
+    constructor(
+        var sessions: Boolean = true,
+        var appLifecycles: Boolean = false,
+        var deepLinks: Boolean = false,
+        var screenViews: Boolean = false,
+        var userInteractions: Boolean = false,
+    ) {
+        // Prebuilt options for easier usage
+        companion object {
+            @JvmField
+            val ALL =
+                DefaultTrackingOptions(
+                    sessions = true,
+                    appLifecycles = true,
+                    deepLinks = true,
+                    screenViews = true,
+                    userInteractions = true,
```

Comment:
[Can we make this opt-in](https://amplitude.slack.com/archives/C072Y92GTCH/p1718121684448949?thread_ts=1717709659.326509&cid=C072Y92GTCH) until we try with a small group of customers?

Reply:
I added an annotation to make`userInteractions` require direct opt-in (https://github.com/amplitude/Amplitude-Kotlin/pull/199/commits/5309b73c7dc7c7e40977c51e7480d78eac0d4c60):
```
@OptIn(ExperimentalAmplitudeFeature::class)
    override fun onCreate() {
        super.onCreate()

        amplitude = Amplitude(
            Configuration(
                apiKey = AMPLITUDE_API_KEY,
                context = applicationContext,
                defaultTracking = DefaultTrackingOptions(userInteractions = true)
            )
        )
    }
}
```

Opt-in is only required if `userInteractions` is set in the constructor or directly, not when other default tracking options are enabled/disbaled.